tests/common/mod.rs

```diff
@@ -0,0 +1,228 @@
+        Self::new()
+    }
+}
+
+impl Module for SeiModule {
+    type ExecT = SeiMsg;
+    type QueryT = SeiQueryWrapper;
+    type SudoT = Empty;
+
+    fn execute<ExecC, QueryC>(
+        &self,
+        api: &dyn Api,
+        storage: &mut dyn Storage,
+        router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        block: &BlockInfo,
+        sender: Addr,
+        msg: Self::ExecT,
+    ) -> AnyResult<AppResponse>
+    where
+        ExecC: Debug + Clone + PartialEq + JsonSchema + DeserializeOwned + 'static,
+        QueryC: CustomQuery + DeserializeOwned + 'static,
+    {
+        match msg {
+            SeiMsg::CreateDenom { subdenom } => {
+                let denom = format!("factory/{}/{}", sender, subdenom);
+                if storage.get(denom.as_bytes()).is_some() {
+                    return Err(anyhow::anyhow!("denom already exists"));
+                }
+
+                storage.set(denom.as_bytes(), sender.to_string().as_bytes());
+
+                Ok(AppResponse {
+                    events: vec![],
+                    data: Some(to_binary(&denom).unwrap()),
+                })
+            }
+            SeiMsg::MintTokens { amount } => {
+                let owner = storage.get(amount.denom.as_bytes());
+                if owner.is_none() || owner.unwrap() != sender.to_string().as_bytes() {
+                    return Err(anyhow::anyhow!(
+                        "Must be owner of coin factory denom to mint"
+                    ));
+                }
+                router.sudo(
+                    api,
+                    storage,
+                    block,
+                    SudoMsg::Bank(BankSudo::Mint {
+                        to_address: sender.to_string(),
+                        amount: vec![amount],
+                    }),
+                )
+            }
+            SeiMsg::BurnTokens { amount } => {
+                let owner = storage.get(amount.denom.as_bytes());
+                if owner.is_none() || owner.unwrap() != sender.to_string().as_bytes() {
+                    return Err(anyhow::anyhow!(
+                        "Must be owner of coin factory denom to burn"
+                    ));
+                }
+                Ok(router
+                    .execute(
+                        api,
+                        storage,
+                        block,
+                        sender,
+                        CosmosMsg::Bank(BankMsg::Burn {
+                            amount: vec![amount],
+                        }),
+                    )
+                    .unwrap())
+            }
+            _ => panic!("Unexpected custom exec msg"),
+        }
+    }
+
+    fn query(
+        &self,
+        _api: &dyn Api,
+        _storage: &dyn Storage,
+        _querier: &dyn Querier,
+        _block: &BlockInfo,
+        request: Self::QueryT,
+    ) -> AnyResult<Binary> {
+        match request.query_data {
+            SeiQuery::ExchangeRates {} => Ok(Binary::default()),
```

Comment:
similarly for queries, we'd want to have helpers for the implementations that end up longer.